packages/backend/src/app.ts

```diff
@@ -367,6 +370,13 @@ export class App {
       }
     )
 
+    koa.use(async (ctx, next: Koa.Next): Promise<void> => {
+      if (!verifyApiSignature(ctx, this.config) && this.config.apiSecret) {
```

Comment:
nit:

Perhaps only add the middleware if `this.config.apiSecret` then always verify in the middleware? 1 less middleware to traverse when not secured and its not like the config will change during runtime.